### PR TITLE
Logs in details panel

### PR DESCRIPTION
**Pull request by Jivings**
Added job logs to the details panel. Loaded async when the specific panel is opened so we're not loading loads of pointless data.

image

### CODE REVIEW

**Comment by Jivings** [comment]
fixes #97

File: src/queueAdapters/bull.ts [hunk 1]
```diff
@@ -25,6 +25,10 @@ export class BullAdapter implements QueueAdapter {
     return this.queue.getJob(id)
   }
 
+  public getLogs(id: string): Promise<string[]> {
+    return this.queue.getJobLogs(id).then(({logs}) => logs);
```
Comment (felixmosh) [inline src/queueAdapters/bull.ts:29]: You have a lint issue here. pls fix

File: src/queueAdapters/bullMQ.ts [hunk 2]
```diff
@@ -28,6 +28,10 @@ export class BullMQAdapter implements QueueAdapter {
     return this.queue.getJob(id)
   }
 
+  public getLogs(id: string): Promise<string[]> {
+    return this.queue.getJobLogs(id).then(({logs}) => logs);
```
Comment (felixmosh) [inline src/queueAdapters/bullMQ.ts:32]: You have a lint issue here. pls fix

File: src/ui/components/Logs/Logs.tsx [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+export const Logs = ({
+  queueName,
+  jobId,
+}: {
+  queueName: string
+  jobId: string | number | undefined
+}) => {
+  const [state, setState] = useState([])
+
+  useEffect(() => {
```
Comment (felixmosh) [inline src/ui/components/Logs/Logs.tsx:14]: Components should not know how to make AJAX requests, it breaks the entire encapsulation.
All "actions" are under `actions` object (which get passed already to JobCard).

This way you don't need to pass the queue name every where.

Pls refactor this 🙏🏼

**Review by felixmosh — CHANGES_REQUESTED** [review]

**Comment by omairvaiyani** [comment]
@Jivings need some help getting this over the line?

**Comment by Jivings** [comment]
@omairvaiyani feel free to take over! I'm in a bad position where I can dedicate the time to fix my own issues but I'm not profecient in typescript, so I can't spend the extra time getting everything perfect.

Right now I have this running on a production bull instance for the last three weeks with no issues 👍

**Pull request closed without merging**